ingestion/src/metadata/ingestion/source/database/clickhouse/lineage.py

```diff
@@ -36,5 +47,134 @@ class ClickhouseLineageSource(ClickhouseQueryParserSource, LineageSource):
     """
 
     database_field = ""
-
     schema_field = "databases"
+
+    def resolve_table_entities_with_fallback(
+        self,
+        metadata: OpenMetadata,
+        service_name: str,
+        database_name: Optional[str],
+        database_schema: Optional[str],
+        table_name: str,
+    ) -> Optional[List[Any]]:
+        """
+        Resolve table entities using Clickhouse-specific fallback strategies.
+
+        Addresses common issue where Clickhouse queries use schema.table format
+        that doesn't match OpenMetadata's database/schema/table hierarchy.
+
+        Args:
+            metadata: OpenMetadata client
+            service_name: Service name for table lookup
+            database_name: Database context
+            database_schema: Schema context
+            table_name: Table name to resolve (may include schema prefix)
+
+        Returns:
+            List of table entities if found, None otherwise
+        """
+        # Define resolution strategies in priority order
+        strategies = [
+            # 1. Original table name (most common case)
+            (database_name, database_schema, table_name, "original"),
+        ]
+
+        # 2. Handle schema.table format (common in Clickhouse)
+        if "." in table_name and table_name.count(".") == 1:
+            schema_part, table_part = table_name.split(".", 1)
+            strategies.extend(
+                [
+                    # Try parsed schema as database_schema
+                    (database_name, schema_part, table_part, "parsed_schema_as_schema"),
+                    # Try parsed schema as database_name (Clickhouse databases-as-schemas)
+                    (
+                        schema_part,
+                        database_schema,
+                        table_part,
+                        "parsed_schema_as_database",
+                    ),
```

Comment:
this situation should never occur right because clickhouse is a single database source and in OpenMetadata we always mark database as `default` by default so schema name can never be at the place of database 🤔

Reply:
I agree. I will remove this strategy.